impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -153,363 +97,168 @@ public synchronized void stop() {
     }
-    throw new CachePersistenceException("Unknown space " + identifier);
-  }
-
-  private PersistenceSpace getPersistenceSpace(PersistenceSpaceIdentifier<?> identifier) {
-    for (PersistenceSpace persistenceSpace : knownPersistenceSpaces.values()) {
-      if (persistenceSpace.identifier.equals(identifier)) {
-        return persistenceSpace;
-      }
-    }
-    return null;
+    SafeSpace ss = ((DefaultSafeSpaceIdentifier) safeSpaceId).safeSpace;
+    destroy(ss, verbose);
   }
 
   /**
    * {@inheritDoc}
    */
-  @Override
-  public FileBasedPersistenceContext createPersistenceContextWithin(PersistenceSpaceIdentifier<?> identifier, String name) throws CachePersistenceException {
-    if (containsSpace(identifier)) {
-      validateName(name);
-      File directory = new File(((DefaultPersistenceSpaceIdentifier) identifier).getDirectory(), name);
-      try {
-        create(directory);
-      } catch (IOException ex) {
-        throw new CachePersistenceException("Unable to create persistence context for " + name + " in " + identifier);
+  public void destroyAll(String owner) {
+    File ownerDirectory = new File(rootDirectory, owner);
+    boolean cleared = true;
+    if (ownerDirectory.exists() && ownerDirectory.isDirectory()) {
+      cleared = false;
+      if (recursiveDeleteDirectoryContent(ownerDirectory)) {
+        LOGGER.debug("Destroyed all file based persistence contexts owned by {}", owner);
+        cleared = ownerDirectory.delete();
       }
-      return new DefaultFileBasedPersistenceContext(directory);
-    } else {
-      throw new CachePersistenceException("Unknown space: " + identifier);
     }
-  }
-
-  private void validateName(String name) {
-    if (!name.matches("[a-zA-Z0-9\\-_]+")) {
-      throw new IllegalArgumentException("Name is invalid for persistence context: " + name);
+    if (!cleared) {
+      LOGGER.warn("Could not delete all file based persistence contexts owned by {}", owner);
     }
   }
 
-  private boolean containsSpace(PersistenceSpaceIdentifier<?> identifier) {
-    for (PersistenceSpace persistenceSpace : knownPersistenceSpaces.values()) {
-      if (persistenceSpace.identifier.equals(identifier)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  File getLockFile() {
-    return lockFile;
+  @Override
+  public void destroySafeSpace(String owner, String name, boolean verbose) {
+    SafeSpace ss = createSafeSpaceLogical(owner, name);
+    destroy(ss, verbose);
   }
 
-  private File getDirectoryFor(String identifier) {
-    File directory = new File(rootDirectory, safeIdentifier(identifier));
-
-    for (File parent = directory.getParentFile(); parent != null; parent = parent.getParentFile()) {
-      if (rootDirectory.equals(parent)) {
-        return directory;
-      }
+  @Override
+  public synchronized void startForMaintenance(ServiceProvider<MaintainableService> serviceProvider) {
+    if (rootDirectory == null) {
```

Comment:
yes. this should be an error. Infact, this check is not even necessary as this object will not exist if rootDirectory is null...